pilot/lib/pilot/build_manager.rb

```diff
@@ -110,25 +110,29 @@ def distribute_build(uploaded_build, options)
 
       # This is where we could add a check to see if encryption is required and has been updated
       uploaded_build.export_compliance.encryption_updated = false
-      uploaded_build.beta_review_info.demo_account_required = false
-      uploaded_build.submit_for_testflight_review!
 
       if options[:distribute_external]
+        uploaded_build.beta_review_info.demo_account_required = false
+        uploaded_build.submit_for_testflight_review!
         external_group = Spaceship::TestFlight::Group.default_external_group(app_id: uploaded_build.app_id)
         uploaded_build.add_group!(external_group) unless external_group.nil?
 
         if external_group.nil? && options[:groups].nil?
           UI.user_error!("You must specify at least one group using the `:groups` option to distribute externally")
         end
 
-      end
-
-      if options[:groups]
-        groups = Spaceship::TestFlight::Group.filter_groups(app_id: uploaded_build.app_id) do |group|
-          options[:groups].include?(group.name)
+        if options[:groups]
```

Comment:
Correct: groups only apply to external distribution.  There is only a single internal distribution group.